### DISCUSSION TITLE
form.Field validator no longer infers the correct TS type?

### DISCUSSION THREAD

**Discussion by jose-biescas** [TanStack/form#1071] [Q&A] [answered]
Recently updated to the latest version of `react-form@0.40.1` and validator functions are no longer inferring the correct type of the field even if a type is passed to the form instance.

Example:

```
type MyType = {
   firstName: string
}

const form = useForm<MyType>({firstName: ''})

...more code

<form.Field
            name="firstName"
            validators={{
              // value here has a type of 'any'
              onChange: ({ value }) =>
                !value
                  ? 'A first name is required'
                  : value.length < 3
                    ? 'First name must be at least 3 characters'
                    : undefined,
              onChangeAsyncDebounceMs: 500,
             // value here has a type of 'any'
              onChangeAsync: async ({ value }) => {
                await new Promise((resolve) => setTimeout(resolve, 1000))
                return (
                  value.includes('error') && 'No "error" allowed in first name'
                )
              },
            }}
            children={(field) => {
              return (
                ...input
              )
            }}
          />
```

The TS errors can even be seen in the provided simple example: https://tanstack.com/form/latest/docs/framework/react/examples/simple

Am I missing something?

**Comment by Balastrong** [comment, marked as answer]
You're right, to cover an edge case I missed in 0.40 it broke the types in 0.41, I apologize.

You can downgrade to 0.40 for now as types should be fine there, while I'll try to fix that (and understand why type tests are still passing)

**Comment by jose-biescas** [comment]
No worries, thank you very much for the quick response!

**Reply by Balastrong** [reply]
It's been a wild night but I think I found it, can you please try with `0.40.2`? Examples in the docs are now working :)
